src/path.cr

```diff
@@ -625,6 +629,17 @@ struct Path
     expanded.normalize(remove_final_separator: false)
   end
 
+  private def resolve_home(home)
+    home = case home
+           when Path   then home
+           when String then Path[home]
+           when true   then Path.home
+           else             raise ArgumentError.new("Expected Path | String | Bool, not #{home.class}")
```

Comment:
This else branch should not raise `ArgumentError` because the method is not callable directly and from the calls in `#expand`, the only possible argument values are either `Path | String` or `true`.

So I'd suggest to either `raise "unreachable"` or remove the entire else branch and replace `true` with `Bool`, because for any boolean value (which in this case can only be `true`), it should use `Path.home`.

Reply:
Also IMO it's a bit more concise to put the assign inside the branches. You can even leave out the `Path` branch.  This requires to omit the `else` branch by matching against `Bool`.

```cr
case home
  when String then home = Path[home]
  when Bool   then home = Path.home
end
```
I'd strongly prefer this notation, but it's not mandatory.

Reply:
`else raise` is to catch future code change changes causing unexpected behavior.  Are you sure you want them gone?